// @aws-sdk/client-secrets-manager/dist-es/models/models_0.js
import { __assign } from "tslib";
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
export var ReplicaRegionType;
(function (ReplicaRegionType) {
    ReplicaRegionType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaRegionType || (ReplicaRegionType = {}));
export var CancelRotateSecretRequest;
(function (CancelRotateSecretRequest) {
    CancelRotateSecretRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CancelRotateSecretRequest || (CancelRotateSecretRequest = {}));
export var CancelRotateSecretResponse;
(function (CancelRotateSecretResponse) {
    CancelRotateSecretResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CancelRotateSecretResponse || (CancelRotateSecretResponse = {}));
export var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Tag || (Tag = {}));
export var CreateSecretRequest;
(function (CreateSecretRequest) {
    CreateSecretRequest.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.SecretBinary && { SecretBinary: SENSITIVE_STRING })), (obj.SecretString && { SecretString: SENSITIVE_STRING }))); };
})(CreateSecretRequest || (CreateSecretRequest = {}));
export var StatusType;
(function (StatusType) {
    StatusType["Failed"] = "Failed";
    StatusType["InProgress"] = "InProgress";
    StatusType["InSync"] = "InSync";
})(StatusType || (StatusType = {}));
export var ReplicationStatusType;
(function (ReplicationStatusType) {
    ReplicationStatusType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicationStatusType || (ReplicationStatusType = {}));
export var CreateSecretResponse;
(function (CreateSecretResponse) {
    CreateSecretResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateSecretResponse || (CreateSecretResponse = {}));
export var DeleteResourcePolicyRequest;
(function (DeleteResourcePolicyRequest) {
    DeleteResourcePolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteResourcePolicyRequest || (DeleteResourcePolicyRequest = {}));
export var DeleteResourcePolicyResponse;
(function (DeleteResourcePolicyResponse) {
    DeleteResourcePolicyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteResourcePolicyResponse || (DeleteResourcePolicyResponse = {}));
export var DeleteSecretRequest;
(function (DeleteSecretRequest) {
    DeleteSecretRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteSecretRequest || (DeleteSecretRequest = {}));
export var DeleteSecretResponse;
(function (DeleteSecretResponse) {
    DeleteSecretResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteSecretResponse || (DeleteSecretResponse = {}));
export var DescribeSecretRequest;
(function (DescribeSecretRequest) {
    DescribeSecretRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeSecretRequest || (DescribeSecretRequest = {}));
export var RotationRulesType;
(function (RotationRulesType) {
    RotationRulesType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RotationRulesType || (RotationRulesType = {}));
export var DescribeSecretResponse;
(function (DescribeSecretResponse) {
    DescribeSecretResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeSecretResponse || (DescribeSecretResponse = {}));
export var Filter;
(function (Filter) {
    Filter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Filter || (Filter = {}));
export var GetRandomPasswordRequest;
(function (GetRandomPasswordRequest) {
    GetRandomPasswordRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetRandomPasswordRequest || (GetRandomPasswordRequest = {}));
export var GetRandomPasswordResponse;
(function (GetRandomPasswordResponse) {
    GetRandomPasswordResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.RandomPassword && { RandomPassword: SENSITIVE_STRING }))); };
})(GetRandomPasswordResponse || (GetRandomPasswordResponse = {}));
export var GetResourcePolicyRequest;
(function (GetResourcePolicyRequest) {
    GetResourcePolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetResourcePolicyRequest || (GetResourcePolicyRequest = {}));
export var GetResourcePolicyResponse;
(function (GetResourcePolicyResponse) {
    GetResourcePolicyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetResourcePolicyResponse || (GetResourcePolicyResponse = {}));
export var GetSecretValueRequest;
(function (GetSecretValueRequest) {
    GetSecretValueRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetSecretValueRequest || (GetSecretValueRequest = {}));
export var GetSecretValueResponse;
(function (GetSecretValueResponse) {
    GetSecretValueResponse.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.SecretBinary && { SecretBinary: SENSITIVE_STRING })), (obj.SecretString && { SecretString: SENSITIVE_STRING }))); };
})(GetSecretValueResponse || (GetSecretValueResponse = {}));
export var SortOrderType;
(function (SortOrderType) {
    SortOrderType["asc"] = "asc";
    SortOrderType["desc"] = "desc";
})(SortOrderType || (SortOrderType = {}));
export var ListSecretsRequest;
(function (ListSecretsRequest) {
    ListSecretsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSecretsRequest || (ListSecretsRequest = {}));
export var SecretListEntry;
(function (SecretListEntry) {
    SecretListEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SecretListEntry || (SecretListEntry = {}));
export var ListSecretsResponse;
(function (ListSecretsResponse) {
    ListSecretsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSecretsResponse || (ListSecretsResponse = {}));
export var ListSecretVersionIdsRequest;
(function (ListSecretVersionIdsRequest) {
    ListSecretVersionIdsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSecretVersionIdsRequest || (ListSecretVersionIdsRequest = {}));
export var SecretVersionsListEntry;
(function (SecretVersionsListEntry) {
    SecretVersionsListEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SecretVersionsListEntry || (SecretVersionsListEntry = {}));
export var ListSecretVersionIdsResponse;
(function (ListSecretVersionIdsResponse) {
    ListSecretVersionIdsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListSecretVersionIdsResponse || (ListSecretVersionIdsResponse = {}));
export var PutResourcePolicyRequest;
(function (PutResourcePolicyRequest) {
    PutResourcePolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutResourcePolicyRequest || (PutResourcePolicyRequest = {}));
export var PutResourcePolicyResponse;
(function (PutResourcePolicyResponse) {
    PutResourcePolicyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutResourcePolicyResponse || (PutResourcePolicyResponse = {}));
export var PutSecretValueRequest;
(function (PutSecretValueRequest) {
    PutSecretValueRequest.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.SecretBinary && { SecretBinary: SENSITIVE_STRING })), (obj.SecretString && { SecretString: SENSITIVE_STRING }))); };
})(PutSecretValueRequest || (PutSecretValueRequest = {}));
export var PutSecretValueResponse;
(function (PutSecretValueResponse) {
    PutSecretValueResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutSecretValueResponse || (PutSecretValueResponse = {}));
export var RemoveRegionsFromReplicationRequest;
(function (RemoveRegionsFromReplicationRequest) {
    RemoveRegionsFromReplicationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RemoveRegionsFromReplicationRequest || (RemoveRegionsFromReplicationRequest = {}));
export var RemoveRegionsFromReplicationResponse;
(function (RemoveRegionsFromReplicationResponse) {
    RemoveRegionsFromReplicationResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RemoveRegionsFromReplicationResponse || (RemoveRegionsFromReplicationResponse = {}));
export var ReplicateSecretToRegionsRequest;
(function (ReplicateSecretToRegionsRequest) {
    ReplicateSecretToRegionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicateSecretToRegionsRequest || (ReplicateSecretToRegionsRequest = {}));
export var ReplicateSecretToRegionsResponse;
(function (ReplicateSecretToRegionsResponse) {
    ReplicateSecretToRegionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicateSecretToRegionsResponse || (ReplicateSecretToRegionsResponse = {}));
export var RestoreSecretRequest;
(function (RestoreSecretRequest) {
    RestoreSecretRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RestoreSecretRequest || (RestoreSecretRequest = {}));
export var RestoreSecretResponse;
(function (RestoreSecretResponse) {
    RestoreSecretResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RestoreSecretResponse || (RestoreSecretResponse = {}));
export var RotateSecretRequest;
(function (RotateSecretRequest) {
    RotateSecretRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RotateSecretRequest || (RotateSecretRequest = {}));
export var RotateSecretResponse;
(function (RotateSecretResponse) {
    RotateSecretResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RotateSecretResponse || (RotateSecretResponse = {}));
export var StopReplicationToReplicaRequest;
(function (StopReplicationToReplicaRequest) {
    StopReplicationToReplicaRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StopReplicationToReplicaRequest || (StopReplicationToReplicaRequest = {}));
export var StopReplicationToReplicaResponse;
(function (StopReplicationToReplicaResponse) {
    StopReplicationToReplicaResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StopReplicationToReplicaResponse || (StopReplicationToReplicaResponse = {}));
export var TagResourceRequest;
(function (TagResourceRequest) {
    TagResourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagResourceRequest || (TagResourceRequest = {}));
export var UntagResourceRequest;
(function (UntagResourceRequest) {
    UntagResourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagResourceRequest || (UntagResourceRequest = {}));
export var UpdateSecretRequest;
(function (UpdateSecretRequest) {
    UpdateSecretRequest.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.SecretBinary && { SecretBinary: SENSITIVE_STRING })), (obj.SecretString && { SecretString: SENSITIVE_STRING }))); };
})(UpdateSecretRequest || (UpdateSecretRequest = {}));
export var UpdateSecretResponse;
(function (UpdateSecretResponse) {
    UpdateSecretResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateSecretResponse || (UpdateSecretResponse = {}));
export var UpdateSecretVersionStageRequest;
(function (UpdateSecretVersionStageRequest) {
    UpdateSecretVersionStageRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateSecretVersionStageRequest || (UpdateSecretVersionStageRequest = {}));
export var UpdateSecretVersionStageResponse;
(function (UpdateSecretVersionStageResponse) {
    UpdateSecretVersionStageResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateSecretVersionStageResponse || (UpdateSecretVersionStageResponse = {}));
export var ValidateResourcePolicyRequest;
(function (ValidateResourcePolicyRequest) {
    ValidateResourcePolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ValidateResourcePolicyRequest || (ValidateResourcePolicyRequest = {}));
export var ValidationErrorsEntry;
(function (ValidationErrorsEntry) {
    ValidationErrorsEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ValidationErrorsEntry || (ValidationErrorsEntry = {}));
export var ValidateResourcePolicyResponse;
(function (ValidateResourcePolicyResponse) {
    ValidateResourcePolicyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ValidateResourcePolicyResponse || (ValidateResourcePolicyResponse = {}));
